https://bugs.gentoo.org/663348

compile fix for  invalid conversion from char to const char*
thx to Michael Brakemeier for point this out

Signed-off-by: Joerg Bornkessel <hd_brummy@gentoo.org> (29 Feb 2020)
diff -Naur vdr-loadepg-0.2.7.orig/loadepg.c vdr-loadepg-0.2.7/loadepg.c
--- vdr-loadepg-0.2.7.orig/loadepg.c	2020-02-28 14:08:35.713790460 +0100
+++ vdr-loadepg-0.2.7/loadepg.c	2020-02-28 14:11:05.457790460 +0100
@@ -162,7 +162,7 @@
             (lProviders + nProviders)->SourceId = cSource::FromString(string2);
             asprintf(&(lProviders + nProviders)->Parm1, "%i:%c:%s:%i", int1, char1, string2, int2);
             asprintf(&(lProviders + nProviders)->Parm2, "%s", string3);
-            asprintf(&(lProviders + nProviders)->Parm3, (char *) '\0');
+            asprintf(&(lProviders + nProviders)->Parm3, (char *) "");
             nProviders ++;
             IsSkyThemesNull = true;
           }
@@ -172,8 +172,8 @@
             (lProviders + nProviders)->DataFormat = DATA_FORMAT_MHW_1;
             (lProviders + nProviders)->SourceId = cSource::FromString(string2);
             asprintf(&(lProviders + nProviders)->Parm1, "%i:%c:%s:%i", int1, char1, string2, int2);
-            asprintf(&(lProviders + nProviders)->Parm2, (char *) '\0');
-            asprintf(&(lProviders + nProviders)->Parm3, (char *) '\0');
+            asprintf(&(lProviders + nProviders)->Parm2, (char *) "");
+            asprintf(&(lProviders + nProviders)->Parm3, (char *) "");
             nProviders ++;
           }
           else if(sscanf(Line, "MHW_2=%[^:] :%i :%c :%[^:] :%i ", string1, &int1, &char1, string2, &int2) == 5)
@@ -182,8 +182,8 @@
             (lProviders + nProviders)->DataFormat = DATA_FORMAT_MHW_2;
             (lProviders + nProviders)->SourceId = cSource::FromString(string2);
             asprintf(&(lProviders + nProviders)->Parm1, "%i:%c:%s:%i", int1, char1, string2, int2);
-            asprintf(&(lProviders + nProviders)->Parm2, (char *) '\0');
-            asprintf(&(lProviders + nProviders)->Parm3, (char *) '\0');
+            asprintf(&(lProviders + nProviders)->Parm2, (char *) "");
+            asprintf(&(lProviders + nProviders)->Parm3, (char *) "");
             nProviders ++;
           }
           else if(sscanf(Line, "FILE=%[^:] :%s ", string1, string2) == 2)
@@ -192,8 +192,8 @@
             (lProviders + nProviders)->DataFormat = DATA_FORMAT_FILE;
             (lProviders + nProviders)->SourceId = 0;
             asprintf(&(lProviders + nProviders)->Parm1, "%s", string2);
-            asprintf(&(lProviders + nProviders)->Parm2, (char *) '\0');
-            asprintf(&(lProviders + nProviders)->Parm3, (char *) '\0');
+            asprintf(&(lProviders + nProviders)->Parm2, (char *) "");
+            asprintf(&(lProviders + nProviders)->Parm3, (char *) "");
             nProviders ++;
           }
           else if(sscanf(Line, "SCRIPT=%[^:] :%[^:] :%s ", string1, string2, string3) == 3)
@@ -203,7 +203,7 @@
             (lProviders + nProviders)->SourceId = 0;
             asprintf(&(lProviders + nProviders)->Parm1, "%s", string2);
             asprintf(&(lProviders + nProviders)->Parm2, "%s", string3);
-            asprintf(&(lProviders + nProviders)->Parm3, (char *) '\0');
+            asprintf(&(lProviders + nProviders)->Parm3, (char *) "");
             nProviders ++;
           }
         }
